cypress/integration/dashboard/dashboard_filtering_spec.js

```diff
@@ -0,0 +1,315 @@
+/**
+ * dashboard_filtering test suite description:
+ * 1) Create a new dashboard, populate with visualizations
+ * 2) Set a filter that excludes all data, and check the visualizations for proper updates
+ * 3) Set the existing filter as pinned, re-check the visualizations
+ * 4) Remove the filter, and check the visualizations for proper updates
+ * 5) Create a new dashboard, and populate with a pie graph
+ * 6) Perform apply different filters on the pie graph and check the visualization for proper updates
+ * 7) Remove all filters and ensure the graph reverts to its original design
+ * 8) Test adding another pie graph to the dashboard and applying a filter to both grpahs
+ */
+
+describe('dashboard filtering', () => {
+  before(() => {
+    // Increase the viewport size to prevent pop-up notifications from blocking UI element
+    // E.g. after adding a saved visualization, a notification appears in the bottom right
+    // which could block clicking on a button located behind it.
+    cy.viewport(1900, 1080)
```

Comment:
No blocker. Just curious where do 1900 and 1080 come from? Is it some pre-determined value for laptop? why can't be `cy.viewport(1440, 900)`?

Reply:
The 1900x1080 viewport size was chosen somewhat arbitrarily to match the Full HD resolution, and a viewport of 1440x900 would also be valid. The default viewport size of 1000x660 was causing UI elements to overlap, and I felt that the 1900x1080 resolution was large enough to prevent any unwanted overlapping. The `cy.viewport` function does have presets available for different device screens, which could be helpful if testing whether certain functionality works the same on different devices (more information can be found [here](https://docs.cypress.io/api/commands/viewport#Arguments)).

Reply:
I see. thanks for the explain.